test/unit/manipulation.js

```diff
@@ -3099,3 +3099,55 @@ testIframe(
 		} );
 	}
 );
+
+// Custom event handler test case
+
+QUnit.test( "should handle custom '_se-custom-destroy' event correctly", function( assert ) {
+
+	// Define the expected number of assertions
+
+	assert.expect( 4 );
+
+	// Set up the HTML structure by appending to the existing fixture
+
+	var fixture = jQuery( "#qunit-fixture" );
+
+	fixture.append( `
+	<div id="container">
+	  <div class="guarded removeself" data-elt="one">
+	    Guarded 1
+	  </div>
+	  <div class="guarded" data-elt="two">
+	    Guarded 2
+	  </div>
+	  <div class="guarded" data-elt="three">
+	    Guarded 3
+	  </div>
+	</div>
+	` );
+
+	// Define the custom event handler
+	jQuery.event.special[ "_se-custom-destroy" ] = {
+	remove: function( _handleObj ) {
+		var $t = jQuery( this );
+		assert.step( $t.data( "elt" ) );
+		if ( $t.is( ".removeself" ) ) {
+			$t.remove();
+		}
+	}
+	};
+
+	// Attach an empty handler to trigger the `remove`
+	// logic for the custom event when the element is removed.
+
+	jQuery( ".guarded" ).on( "_se-custom-destroy", function( ) { } );
+
+	// Trigger the event by emptying the container
```

Comment:
This is not triggering the event; it's triggering its removal logic. Let's put that in words here:
```suggestion
	// Trigger the event's removal logic by emptying the container
```

Reply:
Done